### ISSUE TITLE
"Join" button not working

### ISSUE THREAD

**Issue by ghost** [melvin0008/ionic-socketio-chat-client#7]
Hey there! :)
I am experiencing an issue as the "Join" button is not working / nothing happens.
To reduce the potential of failure I exactly followed your tutorial / instructions on Git.
1. Cloned you repo
2. Changed directory to the project
3. cordova platform add android
4. cordova plugin add cordova-plugin-whitelist
5. cordova plugin add com.ionic.keyboard
6. ionic run android

App starts, but "join" not working.
I also tried "ionic serve", but same problem here.

After some investigations I had success when changing content security to
<meta http-equiv="Content-Security-Policy" content="default-src *">

but I am not sure if this is the correct way of solving the problem?

Thank you in advance!

**Comment by melvin0008** [comment]
Yes, that is one way to do it. But ionic does this for you if you add two statements in your  config.xml

```
<allow-navigation href="*" />
<allow-intent href="http://*/*" />
```

This is not advised for production applications since href is set to wildcard and navigation will be allowed for every link.  
Read this link [https://github.com/apache/cordova-plugin-whitelist#navigation-whitelist](https://github.com/apache/cordova-plugin-whitelist#navigation-whitelist) to learn more about it.

So the href field to be used for production should have the urls you want to point to. 
For this project I would have http://chat.socket.io in the href

Hope this helped.

**Comment by dirkaka** [comment]
Hey @melvin0008 . I tried adding the code you suggested into config.xml but still the join button does not work.

**Comment by sathyakrishna** [comment]
join button is not working
